core/src/test/java/nu/yona/server/analysis/service/ActivityServiceTests.java

```diff
@@ -118,19 +131,135 @@ public void setUp()
 	{
 		return Collections.singletonMap(Locale.forLanguageTag("en-US"), string);
 	}
+	
+	@Test
+	public void dayActivityOverview()
+	{
+		ZonedDateTime today = getDayStartTime(ZonedDateTime.now(userAnonZone));
+		ZonedDateTime yesterday = today.minusDays(1);
+		
+		// gambling goal was created 2 weeks ago, see above
+		// mock some activity in previous week on yesterday 20:58-21:00
+		DayActivity yesterdayRecordedActivity = DayActivity.createInstance(userAnonEntity, gamblingGoal, yesterday);
+		Activity recordedActivity = Activity.createInstance(Date.from(yesterday.plusHours(20).plusMinutes(58).toInstant()), Date.from(yesterday.plusHours(21).plusMinutes(00).toInstant()));
+		yesterdayRecordedActivity.addActivity(recordedActivity);
+		when(mockDayActivityRepository.findAll(userAnonID, 
+				today.minusDays(2).toLocalDate(), today.toLocalDate()))
+						.thenReturn(new HashSet<DayActivity>(
+								Arrays.asList(yesterdayRecordedActivity)));
+		
+		Page<DayActivityOverviewDTO> dayOverviews = service.getDayActivityOverviews(userID, new PageRequest(0, 3));
+		
+		// assert that the right retrieve from database was done
+		verify(mockDayActivityRepository, times(1)).findAll(userAnonID, today.minusDays(2).toLocalDate(), today.toLocalDate());
+		
+		// because the gambling goal was added with creation date two weeks ago, there are multiple days
+		assertThat(dayOverviews.getNumberOfElements(), equalTo(3));
```

Comment:
Doesn't this vary with the moment in the week you run the test? I.e. on Tuesday, this is right, but on Friday, it's 6?